python/lsst/daf/butler/core/registry.py

```diff
@@ -700,39 +700,36 @@ def getRegion(self, dataId):
         raise NotImplementedError("Must be implemented by subclass")
 
     @abstractmethod
-    def selectDataUnits(self, collections, expr, neededDatasetTypes, futureDatasetTypes):
-        r"""Evaluate a filter expression and lists of `DatasetType`\ s and
-        return a set of data unit values.
+    def selectDataUnits(self, collections, expression, neededDatasetTypes, futureDatasetTypes):
+        """Evaluate a filter expression and lists of
+        `DatasetTypes <DatasetType>` and return a set of data unit values.
 
         Returned set consists of combinations of units participating in data
         transformation from ``neededDatasetTypes`` to ``futureDatasetTypes``,
         restricted by existing data and filter expression.
 
         Parameters
         ----------
-        collections : `list` of `str`
-            An ordered `list` of collections indicating the collections to
-            search for Datasets.
-        expr : `str`
-            An expression that limits the `DataUnit`\ s and (indirectly) the
-            Datasets returned.
+        collections : `PreFlightCollections`
+            Object which provides names of the input/output collections.
+        expression : `str`
+            An expression that limits the `DataUnits <DataUnit>` and
+            (indirectly) the Datasets returned.
         neededDatasetTypes : `list` of `DatasetType`
-            The `list` of `DatasetType`\ s whose instances should be included
-            in the graph and limit its extent.
+            The `list` of `DatasetTypes <DatasetType>` whose DataUnits will
+            be included in the returned column set. Output is limited by the
+            the Datasets of these DatasetTypes which already exist in the
+            registry.
         futureDatasetTypes : `list` of `DatasetType`
-            The `list` of `DatasetType`\ s whose instances may be added to the
-            graph later, which requires that their `DataUnit` types must be
-            present in the graph.
+            The `list` of `DatasetTypes <DatasetType>` whose DataUnits will
+            be included in the returned column set. It is expected that
+            Datasets for these DatasetTypes do not exist in the registry,
+            but presently this is not checked.
```

Comment:
Not checking `futureDatasetTypes` can be added as an option indeed. 

For preflight we still need to associate units with DatasetType names, and the query that we run also uses DatasetType  names to find existing input Datasets, so we need some DatasetType information there, maybe just a  name.

Reply:
Sounds like it would make the most sense to have another argument that just adds more explicit DataUnits to the query, while leaving this argument as it is.

Reply:
OK, and I'll leave that for the future tickets :slightly_smiling_face: